pkg/api/server/handler_api.go

```diff
@@ -2,32 +2,51 @@ package server
 
 import (
 	"context"
+	"fmt"
 	"net/http"
+	"runtime"
 
+	"github.com/containers/libpod/pkg/api/handlers/utils"
 	log "github.com/sirupsen/logrus"
 )
 
 // APIHandler is a wrapper to enhance HandlerFunc's and remove redundant code
 func APIHandler(ctx context.Context, h http.HandlerFunc) http.HandlerFunc {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		log.Debugf("APIHandler -- Method: %s URL: %s", r.Method, r.URL.String())
-		if err := r.ParseForm(); err != nil {
-			log.Infof("Failed Request: unable to parse form: %q", err)
-		}
+	return func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			log.Debugf("Recovering APIHandler -- Method: %s URL: %s", r.Method, r.URL.String())
+
+			err := recover()
+			if err != nil {
+				buf := make([]byte, 1<<20)
+				n := runtime.Stack(buf, true)
+				log.Warnf("panic: %v, %s", err, buf[:n])
```

Comment:
Can we make it "Recovering from panic: ..."? It could be worth knowing that the service didn't panic as a whole, especially when reading longer logs.